src/nupic/algorithms/Connections.cpp

```diff
@@ -37,6 +37,8 @@ using namespace std;
 using namespace nupic;
 using namespace nupic::algorithms::connections;
 
+static const Permanence EPSILON = 0.00001;
```

Comment:
This is a tough call. Dealing with floating point math is inherently messy. As far as I know, there's no "right" EPSILON. No matter how large the EPSILON, someone can devise an experiment that causes different results between different floating point implementations. I might be wrong, but I think it makes sense for classes to define their own EPSILON based on how they are used.

I agree that this feels a little gross. Some day we should consider expressing permanences as a C++ `ratio` / Python `Fraction`.